### PR TITLE
Support Scala Native

### PR DESCRIPTION
**Pull request by lolgab**
# Pull Request Checklist

* [x] Have you read through the [contributor guidelines](https://www.playframework.com/contributing)?

## Purpose

Add support for Scala Native

### CODE REVIEW

File: play-json/js-native/src/main/scala/StaticBindingJsNative.scala
Comment (lolgab) [inline play-json/js-native/src/main/scala/StaticBindingJsNative.scala:9]: The code in this file was move from `StaticBinding` in the js implementation.

File: play-json/native/src/main/scala/StaticBinding.scala
Comment (cchantep) [inline play-json/native/src/main/scala/StaticBinding.scala:54]:
```suggestion

    val stringified = if (s == null) "null" else s.flatMap(escaped)}
    
```

File: scripts/test-code.sh
Comment (cchantep) [inline scripts/test-code.sh:5]: Why would it be required?

File: play-json/native/src/main/scala/StaticBinding.scala
Comment (lolgab) [reply play-json/native/src/main/scala/StaticBinding.scala:54]: I did `s""""${s.flatMap(escaped)}""""` to add an extra pair of quotes around `s.flatMap(escaped)`

File: scripts/test-code.sh
Comment (lolgab) [reply scripts/test-code.sh:5]: It's because Scala Native requires more RAM to build the binary with the tests.
`-XX:MinRAMPercentage=90.0` and `-XX:MaxRAMPercentage=90.0` allow us to use the RAM available in the machine (especially in the Github Action executor) without making assumptions about its memory size.

File: play-json/js-native/src/main/scala/StaticBindingJsNative.scala
Comment (BillyAutrey) [reply play-json/js-native/src/main/scala/StaticBindingJsNative.scala:9]: I'm just curious here - what's the point of this restructure?

File: play-json/native/src/main/scala/StaticBinding.scala
Comment (BillyAutrey) [inline play-json/native/src/main/scala/StaticBinding.scala:2]: I figure this is needed to ensure linting works on the new rules. But this isn't really associated with Lightbend anymore. @mkurz we should visit revamping the requirements on copyrights and file headers.

File: play-json/js-native/src/main/scala/StaticBindingJsNative.scala
Comment (lolgab) [reply play-json/js-native/src/main/scala/StaticBindingJsNative.scala:9]: > Ah, I think I see - you're making the underlying functionality common, so that both `js` and `native` `StaticBinding` implementations can access this code?

Exactly, to avoid copy-pasting a lot of code, I put the common code here so it's shared and called back and forth from the Native and JS implementations.

Comment (BillyAutrey) [reply play-json/js-native/src/main/scala/StaticBindingJsNative.scala:9]: A small nit then - I found the `js-native` and `StaticBindingJsNative` names a bit confusing. The name doesn't really convey the intent, and isn't intuitive for further possible expansions.

Comment (lolgab) [reply play-json/js-native/src/main/scala/StaticBindingJsNative.scala:9]: the `js-native` directory is automatically added by `sbt-crossproject`.
About `StaticBindingJsNative` I just added the platforms names to the original file name. If you have a better suggestion I change it. I don't have any better name in mind.. maybe `StaticBindingNonJvm` ?

Comment (BillyAutrey) [reply play-json/js-native/src/main/scala/StaticBindingJsNative.scala:9]: Ah, I had a feeling it was sbt plugin shenanigans causing the package name. Makes sense, given that the subprojects aren't explicitly depending on one another in `build.sbt`.

I think it's really a nit though, I hate to delay for this. Let's leave it as is, and leave better naming for later. This isn't a public API, so it shouldn't affect very many users.

I will say though, `StaticBindingNonJvm` or `StaticBindingNonJvmShared` is a bit more descriptive.

**Review by BillyAutrey — APPROVED** [review]
Approved, though we should track the comments I made.

1. Naming of `StaticBindingJsNative` feels very specific, and doesn't match the intent necessarily.
2. We're adding files and putting outdated Lightbend copyright data at the top, just to satisfy CI. We should do something different in the future.

**Comment by BillyAutrey** [comment]
@mkurz I'll hold off on merging, wanted to give you another day to come back and make sure I didn't miss anything. I'll merge tomorrow if you have no other thoughts.

**Comment by mkurz** [comment]
will take a look tonight

**Comment by lolgab** [comment]
@BillyAutrey I can rename `StaticBindingJsNative` to `StaticBindingNonJvm` right now.

**Comment by BillyAutrey** [comment]
@lolgab up to you! Like I said I feel it's a nit, and an easy one to fix later if we want to discuss it further.

**Comment by lolgab** [comment]
> @lolgab up to you! Like I said I feel it's a nit, and an easy one to fix later if we want to discuss it further.

Since we are waiting for tomorrow either way, it doesn't cost me anything to update it :)

**Comment by lolgab** [comment]
Thank you very much for reviewing and approving the PR!

**Review by mkurz — APPROVED** [review]
LGTM, rebased and [upgraded versions](https://repo1.maven.org/maven2/org/scala-native/nscplugin_3.3.0-RC5/), since we are now on Scala 3.3.0-RC5.

File: play-json/native/src/main/scala/StaticBinding.scala
Comment (mkurz) [reply play-json/native/src/main/scala/StaticBinding.scala:2]: I will do that in a upcoming pr.

**Pull request merged**
